ocis-pkg/oidc/claims.go

```diff
@@ -12,3 +17,52 @@ const (
 	OwncloudUUID      = "ownclouduuid"
 	OcisRoutingPolicy = "ocis.routing.policy"
 )
+
+// SplitWithEscaping splits s into segments using separator which can be escaped using the escape string
+// See https://codereview.stackexchange.com/a/280193
+func SplitWithEscaping(s string, separator string, escapeString string) []string {
+	a := strings.Split(s, separator)
+
+	for i := len(a) - 2; i >= 0; i-- {
+		if strings.HasSuffix(a[i], escapeString) {
+			a[i] = a[i][:len(a[i])-len(escapeString)] + separator + a[i+1]
+			a = append(a[:i+1], a[i+2:]...)
+		}
+	}
+	return a
+}
+
+func ReadStringClaim(path string, claims map[string]interface{}) (string, error) {
+	// happy path
+	value, _ := claims[path].(string)
+	if value != "" {
+		return value, nil
+	}
+
+	// try splitting path at .
+	segments := SplitWithEscaping(path, ".", "\\")
+	subclaims := claims
+	lastSegment := len(segments) - 1
+	for i := range segments {
+		if i < lastSegment {
```

Comment:
Maybe just range until `lastSegment` and handle the other case later? Would be easier to understand than the `if ... else` inside the for loop.